appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceApplicationContainer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) [2020] Payara Foundation and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://github.com/payara/Payara/blob/master/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.microprofile.faulttolerance;
+
+import org.glassfish.api.deployment.ApplicationContext;
+import org.glassfish.api.deployment.DeploymentContext;
+
+import fish.payara.microprofile.connector.MicroProfileApplicationContainer;
+
+public class FaultToleranceApplicationContainer extends MicroProfileApplicationContainer {
+
+    protected FaultToleranceApplicationContainer(DeploymentContext deploymentContext) {
+        super(deploymentContext);
+    }
+
+    @Override
+    public boolean start(ApplicationContext ctx) throws Exception {
+        return true;
+    }
+
+    @Override
+    public boolean stop(ApplicationContext ctx) {
+        return true;
+    }
+
+    @Override
+    public boolean resume() throws Exception {
+        return true;
+    }
+
+    @Override
+    public boolean suspend() {
+        return true;
+    }
```

Comment:
Why override the methods identical to how they are implemented in the super class?

Reply:
These aren't implemented in the superclass which is why they're required. I'm hesitant to implement them in the super class, as thinking abstractly it doesn't make sense to me to have these as the default implementation. I want to remove the instances of "magic" that make the class harder to understand. I can add some javadocs to explain what these methods are doing?

Reply:
Must have become confused by all the new classes. I thought I saw these also being implemented in a base class. So you say these are implemented identical several times for different reasons?

Reply:
The `start` and `stop` methods are stubs in a couple of cases, but the `resume` and `suspend` methods may be stubs everywhere. I'll implement default `resume` and `suspend` methods.